packages/cms-data-sync/src/utils/migration.ts

```diff
@@ -57,9 +43,39 @@ export const migrateFromSquidexToContentfulFactory =
           n += 1;
           logger(`Created entry with id ${id}. (${n}/${data.length})`, 'INFO');
           if (item.status === 'PUBLISHED') {
-            return entry;
+            return createdEntry;
           }
+          return null;
+        };
+
+        const updateExistingEntry = async () => {
+          const contentfulEntry = await contentfulEnvironment.getEntry(id);
+          updateEntryFields(contentfulEntry, payload);
+          const updatedEntry = await contentfulEntry.update();
+          await contentfulRateLimiter.removeTokens(1);
+
+          n += 1;
+          logger(`Updated entry with id ${id}. (${n}/${data.length})`, 'INFO');
+          return updatedEntry;
+        };
+
+        const shouldUpdateEntry = updateEntry || isUpdateModeEnabled;
+
+        try {
+          return shouldUpdateEntry
+            ? await updateExistingEntry()
+            : await createEntry();
         } catch (err) {
+          if (err instanceof Error) {
+            const errorParsed = JSON.parse(err.message);
```

Comment:
These message parsing lines should be wrapped in a try-catch so that if the parse fails then the original error is thrown.

See [here](https://github.com/yldio/asap-hub/blob/master/packages/cms-data-sync/src/utils/assets-and-media.ts#L83-L91) for example.

Reply:
hmm it makes sense, thanks for the reference, I'll change that in a bit